Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/DocumentSchema.h

```diff
@@ -213,6 +213,57 @@ namespace AZ::DocumentPropertyEditor
         AZStd::fixed_string<128> m_name;
     };
 
+    //! Represents an attribute that should resolve to an AZ::TypeId with a string representation.
+    class TypeIdAttributeDefinition final : public AttributeDefinition<AZ::TypeId>
+    {
+    public:
+        inline explicit constexpr TypeIdAttributeDefinition(AZStd::string_view name)
+            : AttributeDefinition<AZ::TypeId>(name)
+        {
+        }
+
+        inline Dom::Value ValueToDom(const AZ::TypeId& attribute) const override
+        {
+            return AZ::Dom::Utils::TypeIdToDomValue(attribute);
+        }
+
+        inline AZStd::optional<AZ::TypeId> DomToValue(const Dom::Value& value) const override
+        {
+            // If we happen to see a type ID marshalled as an AZStd::any, go ahead and bring it in.
+            if (value.IsOpaqueValue())
+            {
+                return AttributeDefinition<AZ::TypeId>::DomToValue(value);
+            }
+            if (value.IsString())
+            {
+                auto typeId = AZ::Dom::Utils::DomValueToTypeId(value);
+                if (typeId.IsNull())
+                {
+                    return {};
+                }
+                return typeId;
+            }
+            return {};
```

Comment:
is this equivalent to returning AZStd::nullopt ?

Reply:
Yep, it's just shorthand for value initializing the return type, which is convenient here.